recipes/cppitertools/all/conanfile.py

```diff
@@ -23,13 +26,13 @@ def _source_subfolder(self):
         return "source_subfolder"
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
+        get(self, **self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        rename(self, extracted_dir, self._source_subfolder)
```

Comment:
Could you add a layout please and use `get(self, **self.conan_data["sources"][self.version], strip_root=True` instead?